### ISSUE TITLE
Additional Nuwani commands & notices

### ISSUE THREAD

**Issue by BeanerLVP** [LVPlayground/playground#532]
1. Since some players share the same IP ranges, I strongly feel we should have a range exception command like we used to. Range bans had IDs in the past (48 in the example below), but I reckon we can simply specify the range itself.

```
<@estroe> !rexception add 48 AlphA[Dx]
<+Boteend> Notice: AlphA[Dx] has been added to the exceptions list of range 0.0.*.*
```
```
<@estroe> !rexception remove 48 Coco_Ping
 <+Boteend> Notice: Coco_Ping has been removed from the exceptions list of range 0.0.*.*
```

2. Implement _!rangenote [range] [note]_ and _!rangeinfo [range]_ commands (similar to !addnote and !why) & remove the nickname requirement from the !ban range syntax. This is because a particular range ban may be put in place due to several players in some cases. This way, we can have a log of all banned players in a single range (but it may require a web page like our ban logs if it exceeds certain number of entries on IRC).

```
!rangeinfo 45
[01/12/14 01:16:24] <+Boteend> Rangeban info of range 0.0.*.*: ID: 45, Exceptions: 0, 
Added: 2013-11-19 19:41, Added by: Beaner, Number of bans: 23,
Last ban: 2013-12-17 11:46, Ban delay: 5 seconds. 
Note: mehmetreyis11, [FBI]MehmetReyis, KahramanReyiz11,
Lorenzomehmetaydass, etc.
```

```
[12/16/13 01:47:19] <@Beaner> !rangenote 46 Spam bots
[12/16/13 01:47:19] <+Boteend> Done.
```

3. Implement notices to be displayed in #LVP.Crew whenever IP/serial banned or range banned players try to access the server. Currently, only the IP/serial of such players is displayed in #lvp.echo, without a join message of course. Easy to miss if crew members are not actively keeping an eye on IRC. Helps with ban evasion detection.

```
<+Boteend> Notice: IP 0.0.0.0 (ID:16, Alex_Borquets) matched with banned range
[40] 0.0.*.*. Nick is not on the exceptions list. Banning Alex_Borquets
```

```
Banned player: [nickname] / [IP and/or serial] attempted to join the server.
```

4. Implement a _!serialexception [add / remove] [nickname]_ to allow certain players to join the server even if a serial is banned. This is mainly due to serials not being unique; some are quite common and shared by multiple players (check the topic in the crew board for examples). We may miss out on opportunities to gain new players, or inconvenience our regular players without an option like this.

5. Implement a !clear command on IRC, similar to the /clear command in-game, to clear the main chat in case of highly inappropriate language and/or server advertisements / spam.

6. Implement an !ingamecrew command _(aliases: !admin, !admins, !staff, !crew)_ which has two kinds of outputs, depending on the channel it is used in:

- #lvp.crew & #lvp.management 

>  <+Saber> * Temporary admins: XYZ (Granted by: Holsje), ABC (Granted by: Ricky), _Name3 (Granted by), and so on._
> <+Saber>  * Undercover: Noob1 (Ricky92), Noob2 (Holsje), _Name3 (Main nickname), and so on._
>  <+Saber> * In-game crew: PigBenis (ID: 3), Russell (ID: 5) //formatted with colors based on user levels.

- All other public channels:

>  <+Saber> * In-game crew: PigBenis (ID: 3), Russell (ID: 5)
>  <+Saber> Temporary admins: XYZ (ID: 4) //with "Granted by: Name" being hidden in this case.

7. Implement notices (#lvp.crew & #lvp.echo) when players receive temporary rights; when temporary admins leave the server or have their rights revoked; when a crew member logs in with an undercover nickname; and when an undercover crew member leaves the game. 

>  <+Saber> [13:51:08] * Info: [BB]zkippi has been given temporary administrator rights by [BB]Ricky92. //!givetempadmin
>  <+Saber> [15:15:52] * Info: [BB]zkippi is no longer ingame. //quit or disconnect
>  <+Saber> [13:54:06] * Info: WeedGuy has logged in as PigBenis. //with level based color formatting.
>  <+Saber> [14:03:03] * Info: WeedGuy (PigBenis) is no longer ingame. //with level based color formatting.

>Additionally: [BB]zkippi's administrator rights have been revoked by [BB]Ricky92.

_The same notices should also be sent when !givetempadmin & !takeadmin are implemented in Nuwani._

8. Forward modlogin notices to #lvp.crew in addition to what is currently displayed in #lvp.echo:

>*Nuwini:%#lvp.echo* *** marty_byrde (Id:1) has logged in as Beaner (manager) using /modlogin.
>*Nuwini:%#lvp.echo* *** marty_byrde (Beaner) is no longer in-game.

I'm probably asking for a lot here, but from my past experience, I've found 1, 2, 3, 6, 7, and 8 to be quite useful; and I feel 4 & 5 would be good additions to improve the overall player experience.

**Comment by RussellLVP** [comment]
1. I think this might be doable. Will have to think about how it works exactly. Note that this functionality has never been supported by LVP, it was provided through Boteend on IRC.
2. Our infrastructure doesn't really allow for this information to be stored that way, because of the structured way bans (& notes) are stored.
3. I can do this when (1) is done.
4. I really don't think this is a good idea, serials are a poor way of identifying an individual user. They're also really easy to spoof. How frequently would this have been useful?
5. Tracked in another issue.
6. I don't think we should do this. It's a severe complication of infrastructure, as none of that information is known in the right places. Note that this, for exactly that reason, was therefore never provided by LVP directly, but rather by bots. (Saber, in this case.)
7. Can do this too.
8. Can do this too.

**Comment by RussellLVP** [comment]
tl;dr -

1. Support exceptions to range bans (second table w/ JOIN). Broadcast to `#LVP.crew` when somebody tried to join whilst subject to a range ban.
2. Notices to `#LVP.crew` for: (a) granting temp rights, (b) revoking temp rights, (c) somebody with temp rights leaving the server, (d) usage of `/modlogin`.

**Comment by BeanerLVP** [comment]
Alright, we can forget about 2 & 4. 
6 shouldn't really be required if 7 & 8 are implemented.
